### DISCUSSION TITLE
AV1 AOMEnc denoise-noise-level produces rainbow pixels

### DISCUSSION THREAD

**Discussion by Flyk3** [staxrip/staxrip#1402] [General]
Hi,
I have been experimenting with AV1 and various settings from different sources over the last few months. I tested with scenes from various videos and was happy with the following settings:
`--passes=2 --pass=1 --cq-level=22 --verbose --bit-depth=10 --ivf --threads=2 --lag-in-frames=48 --end-usage=q --enable-fwd-kf=1 --kf-min-dist=12 --kf-max-dist=300 --cpu-used=4 --auto-alt-ref=1 --sharpness=1 --static-thresh=0 --enable-tpl-model=1 --arnr-strength=4 --tune=ssim --enable-qm=1 --quant-b-adapt=1 --aq-mode=1 --denoise-noise-level=8 --enable-dnl-denoising=0`

In the small scenes I used for testing, everything worked perfectly, but after I had finished the first complete films, the following error occurred: There are these rainbow colours in every second frame:
![Encode complete](https://github.com/staxrip/staxrip/assets/145803606/079b3dda-e031-4cb7-91cc-6690d666f2a5)

I then encoded the film again with the standard StaxRip AOMEnc settings and everything looks normal:
![Encode part](https://github.com/staxrip/staxrip/assets/145803606/8082694a-7812-4288-9940-b5f87d5bc2d5)

So I started an encoding for every single setting that is different from the default setting and found out that it is due to the `denoise-noise-level=8`.

Why does this option cause errors? I have tried this with various films and the error does not always occur. I am very surprised, as this option is often recommended on the Internet and many people are happy with it. If I mux the scene above from the original film and then encode it, the error doesn't happen either. It only occurs if I encode the entire film in one go.

Is this perhaps a bug in StaxRip? It is very difficult to test this because encoding takes 4-5 days. Has anyone seen this before?

**Comment by Dendraspis** [comment]
Which tool/player did you use to verify those issues?
And which GPU do you have?  AMD or Nvidia?

**Reply by Flyk3** [reply]
Media Player Classic, VLC and Kodi. I took the screenshots with the Video Comparison Tool from StaxRip.
Nvidia RTX 3080 with the newest drivers.

**Reply by Dendraspis** [reply]
It is also visible in StaxRips Video Comparison?   In that case it seems to be encoder related.

> Is this perhaps a bug in StaxRip?

Definitely not as StaxRip is just a GUI. When you used the same filters/plugins for both encodes, then it's likely the encoders fault. In that case you have to head over to https://aomedia.issues.chromium.org/issues?q=status:open and report the issue to them.
